core/src/gl/extension.cpp

```diff
@@ -0,0 +1,59 @@
+#include "extension.h"
+
+#include <cstring>
+#include <sstream>
+#include <algorithm>
+#include <iterator>
+#include "platform.h"
+#include "gl.h"
+
+namespace Tangram {
+namespace GLExtensions {
+
+bool supportsMapBuffer = false;
+bool supportsVAOs = false;
+static char* s_glExtensions;
+
+bool isAvailable(std::string _extension) {
+    return bool(s_glExtensions)
+      ? strstr(s_glExtensions, _extension.c_str()) != nullptr
+      : false;
+}
+
+void printAvailableExtensions() {
+    if (s_glExtensions == NULL) {
+        LOGW("Extensions string is NULL");
+        return;
+    }
+
+    std::string exts(s_glExtensions);
+    std::stringstream ss(exts);
+
+    ss >> std::noskipws;
+    std::string extension;
+    LOG("GL Extensions available: ");
+    while (std::getline(ss, extension, ' ')) {
+        LOG("\t - %s", extension.c_str());
```

Comment:
True. We were talking about this yesterday. Maybe it makes sense to put it
under debug log method.
On Oct 21, 2015 1:21 AM, "Matt Blair" <notifications@github.com> wrote:

> In core/src/gl/extension.cpp
> <https://github.com/tangrams/tangram-es/pull/352#discussion_r42584171>:
>
> > +}
> > +
> > +void printAvailableExtensions() {
> > +    if (s_glExtensions == NULL) {
> > +        LOGW("Extensions string is NULL");
> > +        return;
> > +    }
> > +
> > +    std::string exts(s_glExtensions);
> > +    std::stringstream ss(exts);
> > +
> > +    ss >> std::noskipws;
> > +    std::string extension;
> > +    LOG("GL Extensions available: ");
> > +    while (std::getline(ss, extension, ' ')) {
> > +        LOG("\t - %s", extension.c_str());
>
> Definitely handy to have this for debugging, but it's a bit noisy to log
> these for every run
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/352/files#r42584171>.
>